test/pageobjects/tallestbuildingitem.page.js

```diff
@@ -0,0 +1,24 @@
+import Item from "../pageutils/item";
+
+class TallestBuildingItem extends Item {
+    constructor(locatorTemplates, itemNumber) {
+        super(locatorTemplates, itemNumber);
+        this.nameLink = $(this.locators["nameLink"]);
```

Comment:
I find it somewhat confusing that this.locators is really implemented in the `Item` class and only comes from the super call here.

It's confusing because I see `this.locators` here, but nothing about where this is implemented when I'm looking at this file. I could guess that it comes from Item, but I'd have to go look there to find this out which would break my concentration. 

But that is me. Why did you decide to implement it this way?

Reply:
@m-schrepel, this.locators is nothing specific but a dictionary to store different locators to locate different values for an item within the container. Let's say there are a table tag with multiple rows. Table will be a container and each row will be an item. Each row has multiple columns. Therefore, you need different locators to locate each value of each column. In this case this.locators from Item class will hold these locators to locate column value.